cogs/birthday/birthday.py

```diff
@@ -433,12 +434,42 @@ async def getSelfBirthday(self, ctx: Context):
                 if month and day:
                     birthday = date(2020, month, day)
                     birthdayStr = "{0:%B} {0:%d}".format(birthday)
-                    await ctx.author.send(
-                        f"{headerGood}: Your birthday is "
-                        f"{spoiler(bold(birthdayStr, escape_formatting=False), escape_formatting=False)}."
-                    )
-                    return
-
+                    try:
+                        await ctx.author.send(
+                            f"{headerGood}: Your birthday is "
+                            f"{spoiler(bold(birthdayStr, escape_formatting=False), escape_formatting=False)}."
+                        )
+                        return
+                    except discord.Forbidden:
+
+                        def check(msg: discord.Message):
+                            return msg.author == ctx.author and msg.channel == ctx.channel
+
+                        await ctx.send(
+                            f"{headerWarn}: I would like to DM you your birthday but it seeems that "
+                            "you have disabled DMs from this server. Would you still like to continue here? "
+                            "Your birthday will be sent here and deleted after a short delay. "
+                            f"\nType {bold('`yes`', escape_formatting=False)} to confirm. "
+                            "\nAnything else will be treated as no."
+                        )
+                        try:
+                            response = await self.bot.wait_for(
+                                "message", timeout=30.0, check=check
+                            )
+                        except asyncio.TimeoutError:
+                            await ctx.send(f"{headerBad}: No response detected. Aborting.")
+                            return
+
+                        if response.content.lower() != "yes":
+                            await ctx.send(f"{headerBad}: Aborting.")
+                            return
+
+                        await ctx.send(
+                            f"{headerGood}: Your birthday is "
+                            f"{spoiler(bold(birthdayStr, escape_formatting=False), escape_formatting=False)}.",
+                            delete_after=5,
+                        )
```

Comment:
Can extract this to an outer variable so that we can use it here and in the `try` block without repeating the same message content.

```suggestion
                        await ctx.send(birthdayInfoMsg, delete_after=5)
```

Reply:
Duplicate string removed in mainflow refactor